contracts/src/helpers/PBHExternalNullifier.sol

```diff
@@ -25,25 +30,33 @@ library PBHExternalNullifier {
     /// nullifier pbhNonce >= numPbhPerMonth
     error InvalidPbhNonce();
 
+    uint8 public constant V1 = 1;
+
     /// @notice Encodes a PBH external nullifier using the provided year, month, and nonce.
     /// @param pbhNonce An 8-bit nonce value (0-255) used to uniquely identify the nullifier within a month.
     /// @param month An 8-bit 1-indexed value representing the month (1-12).
     /// @param year A 16-bit value representing the year (e.g., 2024).
     /// @return The encoded PBHExternalNullifier.
-    function encode(uint8 pbhNonce, uint8 month, uint16 year) internal pure returns (uint256) {
+    function encode(uint8 version, uint8 pbhNonce, uint8 month, uint16 year) internal pure returns (uint256) {
```

Comment:
If `V1` is constant, we can probably omit `version` from the args and encode the external nullifier with `V1` directly.

Reply:
it's true but it makes the API asymmetric and makes testing a little uglier. I could add an `encodeV1` function though?

Reply:
I'll leave it as is. I'm open to changing it later